src/write_gcm.c

```diff
@@ -108,6 +113,10 @@ static const char json_content_type_header[] = "Content-Type: application/json";
 //==============================================================================
 //==============================================================================
 
+static _Bool wg_end_to_end_test_mode() {
+  return atoi(global_option_get("ReadThreads")) == -1;
```

Comment:
Yeah, OK. The issue I faced was how to communicate to the plugins the message that we are in "-T mode". Choices:
1. add a new global variable
2. add a new global_option
3. peek at an existing global_option

I used approach 3. Do you think approach 2 would be better? Or do you have another suggestion?